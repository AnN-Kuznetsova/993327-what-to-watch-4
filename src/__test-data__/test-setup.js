import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

Enzyme.configure({
  adapter: new Adapter(),
});

const noop = () => {};
Object.defineProperty(window, `scrollTo`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `play`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `pause`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `load`, {value: noop, writable: true});
