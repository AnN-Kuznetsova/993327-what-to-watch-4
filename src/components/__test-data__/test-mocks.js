const promoMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: new Date(2014, 0),
  posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundUrl: `img/bg-the-grand-budapest-hotel.jpg`,
};

const movieTitles = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`,
];

const genreName = `Kids & Family`;

const movieTitle = `Aviator`;


export {
  genreName,
  movieTitle,
  movieTitles,
  promoMovie,
};
