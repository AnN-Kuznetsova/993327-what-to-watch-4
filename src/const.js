const NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP = 8;
const NUMBER_OF_CARDS_TO_INCREMENT = 8;
const MAX_FILTER_ELEMENTS_COUNT = 10;
const NUMBER_OF_SIMILAR_FILMS = 4;
const VISIBLE_PARTICIPANTS_COUNT = 4;
const DELAY_PLAYBACK_PREVIEW = 1000;
const MIN_REVIEW_TEXT_LENGTH = 50;
const MAX_REVIEW_TEXT_LENGTH = 400;
const RATING_RANGE = 5;
const ERROR_COLOR = `#a8421e`;

const BASENAME_FOR_DEPLOY = `/htmlacademy-what-to-watch`;

const AppRoute = {
  MAIN: BASENAME_FOR_DEPLOY + `/`,
  SIGN_IN: BASENAME_FOR_DEPLOY + `/login`,
  MY_LIST: BASENAME_FOR_DEPLOY + `/myList`,
  FILM: BASENAME_FOR_DEPLOY + `/films/:id`,
  ADD_REVIEW: BASENAME_FOR_DEPLOY + `/films/:id/review`,
  PLAYER: BASENAME_FOR_DEPLOY + `/player/:id`,
};


export {
  DELAY_PLAYBACK_PREVIEW,
  ERROR_COLOR,
  MAX_FILTER_ELEMENTS_COUNT,
  MAX_REVIEW_TEXT_LENGTH,
  MIN_REVIEW_TEXT_LENGTH,
  NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP,
  NUMBER_OF_CARDS_TO_INCREMENT,
  NUMBER_OF_SIMILAR_FILMS,
  RATING_RANGE,
  VISIBLE_PARTICIPANTS_COUNT,
  AppRoute,
  BASENAME_FOR_DEPLOY,
};
