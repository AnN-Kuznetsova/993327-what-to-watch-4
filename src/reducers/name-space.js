export const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
};
