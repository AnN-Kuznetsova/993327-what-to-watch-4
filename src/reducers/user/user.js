import {extend} from "../../utils/utils";


const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  loginError: null,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_LOGIN_ERROR: `SET_LOGIN_ERROR`,
};

const ActionCreator = {
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  setLoginError: (error) => ({
    type: ActionType.SET_LOGIN_ERROR,
    payload: error,
  }),
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.email,
      password: authData.password,
    })
    .then(() => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.setLoginError(null));
    })
    .catch((error) => {
      dispatch(ActionCreator.setLoginError(error));
    });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.SET_LOGIN_ERROR:
      return extend(state, {
        loginError: action.payload
      });

    default:
      return state;
  }
};


export {
  ActionCreator,
  ActionType,
  AuthorizationStatus,
  Operation,
  reducer,
};
