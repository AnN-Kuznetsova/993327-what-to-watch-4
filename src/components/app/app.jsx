import PropTypes from "prop-types";
import React from "react";
import {Switch, Route, BrowserRouter} from "react-router-dom";
import {connect} from "react-redux";

import {ActionCreator as ApplicationActionCreator} from "../../reducers/application/application";
import {ActionCreator as DataActionCtrator, Operation} from "../../reducers/data/data";
import {ErrorPage} from "../error-page/error-page";
import {MainPage} from "../main-page/main-page";
import {MovieDetailsPage} from "../movie-details-page/movie-details-page";
import {MoviePropType} from "../../prop-types";
import {PageType, NUMBER_OF_SIMILAR_FILMS} from "../../const";
import {PlayerPage} from "../player-page/player-page";
import {SignIn} from "../sign-in/sign-in";
import {getActivePage, getActiveMovie} from "../../reducers/application/selectors";
import {getError} from "../../reducers/data/selectors";


const AppComponent = (props) => {
  const {
    dataError,
    activePage,
    activeMovie,
    onOpenMovieDetailsPage,
  } = props;

  const renderPage = () => {
    window.scrollTo(0, 0);

    switch (activePage) {
      case PageType.MAIN:
        return (
          <MainPage
            promoMovie={activeMovie}
            openMovieDetailsPage={onOpenMovieDetailsPage}
          />
        );
      case PageType.MOVIE_DETAILS:
        return (
          <MovieDetailsPage
            activeMovie={activeMovie}
            onSmallMovieCardClick={onOpenMovieDetailsPage}
          />
        );
      case PageType.PLAYER:
        return (
          <PlayerPage
            movie={activeMovie}
          />
        );
      case PageType.SIGN_IN:
        return (
          <SignIn />
        );
      case PageType.ERROR:
        return (
          <ErrorPage
            dataError={dataError}
          />
        );
      default:
        return null;
    }
  };

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          {renderPage()}
        </Route>
        {/* <Route exact path="/movie-details">
          <MovieDetailsPage
            activeMovie={activeMovie}
            onSmallMovieCardClick={this.openMovieDetailsPage.bind(this)}
          />
        </Route>
        <Route exact path="/sign-in">
          <SignIn onSubmit={() => {}} />
        </Route>*/}
      </Switch>
    </BrowserRouter>
  );
};


AppComponent.propTypes = {
  dataError: PropTypes.object,
  activePage: PropTypes.string.isRequired,
  activeMovie: MoviePropType,
  onOpenMovieDetailsPage: PropTypes.func.isRequired,
};


const mapStateToProps = (state) => ({
  dataError: getError(state),
  activePage: getActivePage(state),
  activeMovie: getActiveMovie(state),
});

const mapDispatchToProps = (dispatch) => ({
  onOpenMovieDetailsPage(movie) {
    dispatch(DataActionCtrator.setMaxMoviesCount(NUMBER_OF_SIMILAR_FILMS));
    dispatch(ApplicationActionCreator.changeActiveMovie(movie));
    dispatch(Operation.loadActiveMovieComments(movie.id));
    dispatch(ApplicationActionCreator.changeGenre(movie.genres[0]));
    dispatch(ApplicationActionCreator.changeActivePage(PageType.MOVIE_DETAILS));
  },
});

const App = connect(mapStateToProps, mapDispatchToProps)(AppComponent);


export {
  AppComponent,
  App,
};
