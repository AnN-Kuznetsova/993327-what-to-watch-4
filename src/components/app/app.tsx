import * as React from "react";
import {Switch, Route, Router, Redirect} from "react-router-dom";
import {connect} from "react-redux";

import {ActionCreator as ApplicationActionCreator} from "../../reducers/application/application";
import {ActionCreator as DataActionCtrator} from "../../reducers/data/data";
import {AddReviewPageWithNewReview} from "../add-review-page/add-review-page";
import {ErrorPage} from "../error-page/error-page";
import {MainPage} from "../main-page/main-page";
import {MovieDetailsPage} from "../movie-details-page/movie-details-page";
import {MovieType, PageType} from "../../types";
import {MyListPage} from "../my-list-page/my-list-page";
import {AppRoute} from "../../const";
import {PlayerPage} from "../player-page/player-page";
import {PrivateRoute} from "../private-route/private-route";
import {RedirectToMainRoute} from "../redirect-to-main-route/redirect-to-main-route";
import {SignIn} from "../sign-in/sign-in";
import {getActivePage} from "../../reducers/application/selectors";
import {getDataError, getMovies, getPromoMovie} from "../../reducers/data/selectors";
import {history} from "../../history";



interface Props {
  dataError?: object;
  activePage: string;
  movies?: MovieType[];
  promoMovie?: MovieType;
  onError: (error: object) => void;
}


class AppComponent extends React.PureComponent<Props, {}> {
  renderPage() {
    const {
      dataError,
      activePage,
      movies,
      promoMovie,
      onError,
    } = this.props;

    window.scrollTo(0, 0);

    switch (activePage) {
      case PageType.MAIN: {
        if (movies && promoMovie) {
          return (
            <MainPage />
          );
        }
        onError({status: 400});
        return (<Redirect to={AppRoute.MAIN} />);
      }

      case PageType.ERROR:
        return (
          <ErrorPage dataError={dataError} />
        );

      default:
        return null;
    }
  }

  render() {
    const {
      onError
    } = this.props;

    return (
      <Router history={history}>
        <Switch>
          <Route exact path={AppRoute.MAIN}>
            {this.renderPage()}
          </Route>

          <RedirectToMainRoute exact path={AppRoute.FILM}
            render={(routeProps) => {
              return (
                <MovieDetailsPage routeProps={routeProps} />
              );
            }}
          />

          <RedirectToMainRoute exact path={AppRoute.PLAYER}
            render={(routeProps) => {
              return (
                <PlayerPage routeProps={routeProps} />
              );
            }}
          />

          <RedirectToMainRoute exact path={AppRoute.SIGN_IN}
            render={() => {
              return (<SignIn />);
            }}
          />

          <PrivateRoute exact path={AppRoute.ADD_REVIEW}
            render={(routeProps) => {
              return (
                <AddReviewPageWithNewReview routeProps={routeProps} />
              );
            }}
          />

          <PrivateRoute exact path={AppRoute.MY_LIST}
            render={() => {
              return (
                <MyListPage />
              );
            }}
          />

          <Route
            render={() => {
              onError({status: 404});
              return (<Redirect to={AppRoute.MAIN} />);
            }}
          />
        </Switch>
      </Router>
    );
  }
}


const mapStateToProps = (state) => ({
  dataError: getDataError(state),
  activePage: getActivePage(state),
  movies: getMovies(state),
  promoMovie: getPromoMovie(state),
});

const mapDispatchToProps = (dispatch) => ({
  onError(error) {
    dispatch(DataActionCtrator.setDataError(error));
    dispatch(ApplicationActionCreator.changeActivePage(PageType.ERROR));
  }
});

const App = connect(mapStateToProps, mapDispatchToProps)(AppComponent);


export {
  AppComponent,
  App,
};
