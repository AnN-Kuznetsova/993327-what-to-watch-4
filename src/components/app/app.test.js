import React from "react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";

import {AppComponent} from "./app";
import {AuthorizationStatus} from "../../reducers/user/user";
import {NameSpace} from "../../reducers/name-space";
import {PageType} from "../../const";

import {mockPromoMovie, mockMovies} from "../../__test-data__/test-mocks";


global.window = Object.create(window);

const nodeMock = {
  createNodeMock: () => {
    return {};
  }
};

const mockStore = configureStore([]);


describe(`Render App`, () => {
  it(`Should match with snapshot`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        movies: mockMovies,
      },
      [NameSpace.APPLICATION]: {
        genre: `All genres`,
        visibleMoviesCount: 8,
        activeMovie: mockPromoMovie,
        activePage: PageType.MAIN,
        prevPage: ``,
        playerStartTime: 0,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });

    const props = {
      dataError: null,
      activePage: PageType.MAIN,
      promoMovie: mockPromoMovie,
      movies: mockMovies,
      authorizationStatus: AuthorizationStatus.AUTH,
      onOpenMovieDetailsPage: () => {},
      onAddReviewButtonClick: () => {},
      setDataError: () => {},
      sendReview: () => {},
      changeActivePage: () => {},
      changeActiveMovie: () => {},
    };

    const appSnapshot = renderer.create(
        <Provider store={store}>
          <AppComponent {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(appSnapshot).toMatchSnapshot();
  });
});
