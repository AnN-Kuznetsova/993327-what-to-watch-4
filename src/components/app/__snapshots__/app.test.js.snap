// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Should match with snapshot 1`] = `
Array [
  <section
    className="movie-card"
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/myList"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="/img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <a
          href="/films/1"
          onClick={[Function]}
        >
          <div
            className="movie-card__poster"
          >
            <img
              alt="The Grand Budapest Hotel"
              height="327"
              src="img/the-grand-budapest-hotel-poster.jpg"
              width="218"
            />
          </div>
        </a>
        <div
          className="movie-card__desc"
        >
          <a
            href="/films/1"
            onClick={[Function]}
            style={
              Object {
                "color": "inherit",
                "textDecoration": "none",
              }
            }
          >
            <h2
              className="movie-card__title"
            >
              The Grand Budapest Hotel
            </h2>
          </a>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Biography
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Detective
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Military
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            History
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Fiction
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thriller
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Adventure
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                className="player__video"
                poster="img/bohemian-rhapsody.jpg"
              />
              <img
                alt="Bohemian Rhapsody"
                height="175"
                src="img/bohemian-rhapsody.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                className="player__video"
                poster="img/aviator.jpg"
              />
              <img
                alt="Aviator"
                height="175"
                src="img/aviator.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                className="player__video"
                poster="img/revenant.jpg"
              />
              <img
                alt="Revenant"
                height="175"
                src="img/revenant.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Revenant
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Render App Should match with snapshot when authorizationStatus is NO_AUTH 1`] = `
Array [
  <section
    className="movie-card"
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header false false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <a
          href="/films/1"
          onClick={[Function]}
        >
          <div
            className="movie-card__poster"
          >
            <img
              alt="The Grand Budapest Hotel"
              height="327"
              src="img/the-grand-budapest-hotel-poster.jpg"
              width="218"
            />
          </div>
        </a>
        <div
          className="movie-card__desc"
        >
          <a
            href="/films/1"
            onClick={[Function]}
            style={
              Object {
                "color": "inherit",
                "textDecoration": "none",
              }
            }
          >
            <h2
              className="movie-card__title"
            >
              The Grand Budapest Hotel
            </h2>
          </a>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Biography
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Detective
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Military
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            History
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Fiction
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thriller
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Adventure
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                className="player__video"
                poster="img/bohemian-rhapsody.jpg"
              />
              <img
                alt="Bohemian Rhapsody"
                height="175"
                src="img/bohemian-rhapsody.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                className="player__video"
                poster="img/aviator.jpg"
              />
              <img
                alt="Aviator"
                height="175"
                src="img/aviator.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                className="player__video"
                poster="img/revenant.jpg"
              />
              <img
                alt="Revenant"
                height="175"
                src="img/revenant.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Revenant
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Render App Should match with snapshot with data error 400 1`] = `
Array [
  <section
    className="movie-card"
    style={
      Object {
        "backgroundColor": "#180202",
      }
    }
  >
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header false false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The request failed
          </h2>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Render App Should match with snapshot with data error 404 1`] = `
Array [
  <section
    className="movie-card"
    style={
      Object {
        "backgroundColor": "#180202",
      }
    }
  >
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header false false"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link false"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Page not found.
          </h2>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
